libraries/Wire/Wire.cpp

```diff
@@ -273,6 +273,16 @@ bool TwoWire::cfg_pins(int max_index) {
 
 /* -------------------------------------------------------------------------- */
 void TwoWire::begin(void) {
+/* -------------------------------------------------------------------------- */  
+  end();
```

Comment:
Is end() needed here? Should we also add it to `begin(address)` ?

Reply:
end() is called in both cases -> in case of master initialization in the line you spotted, in case of slave initialization is called only in the version of begin(uint16_t address) (line 413), because the other 2 begin functions (i.e. begin(int address) and begin(uint8_t address) are in the end calling the version with uint16_t as argument.